ioeffect/src/main/scala/scalaz/ioeffect/MonadIOInstances.scala

```diff
@@ -0,0 +1,81 @@
+package scalaz.ioeffect
+
+import scalaz.Scalaz._
+import scalaz._
+
+abstract class MonadIOInstances {
+  import Isomorphism._
+
+  def fromIso[F[_], G[_]](D: F <~> G)(implicit E: MonadIO[G]): MonadIO[F] =
+    new IsomorphismMonadIO[F, G] {
+      override def G: MonadIO[G] = E
+      override def iso: F <~> G  = D
+    }
+
+  implicit def ioMonadIO[E]: MonadIO[IO[E, ?]] = new MonadIO[IO[E, ?]] {
```

Comment:
move this to a parent `abstract class` and add an `implicit val` for `Task` (caches the instance, and has higher implicit priority)